design/proposals/dbt-delete-merge-schema-branch.md

```diff
@@ -0,0 +1,66 @@
+# Add DBT delete and merge branch-schema operations
+
+---
+
+## Background
+
+To use dbt, users must have an already initialized schema under their dbt profile's target.    
+When using dbt alongside lakeFS, the schema above should have a branch's URI as its location.
+When the users want to branch out and run dbt over a different branch, they first need to create a branch to work on and then create a dbt schema dedicated for that branch (that might have the same name of the branch [by default] or a custom one). #2988 added an option to use `lakectl dbt create-branch-schema --create-branch` to create the dedicated branch and schema at once.
+
+After running dbt over that branch and conducting tests and validations we would want to either: 
+- delete the branch and the schema associated with it
+- merge them into master/main/production/other branch
+- both.
+
+## Goals 
+- Enable deletion of schemas created for dbt branches using lakectl.
+- Enable merge of schemas created for dbt branches using lakectl.
+
+## Non-goals
+
+- Couple branch delete and schema delete operations.
+- Couple branch merge and schema merge operations.
+
+## Suggestions
+
+- `delete-branch-schema` will allow deleting a schema based on a branch name.
+- `merge-branch-schema` will allow merging a schema to a destination schema by branch names.
+
+### `delete-branch-schema` Usage
+
+`lakectl dbt delete-branch-schema --branch <branch name>`: Delete the schema associated with the input branch name.
+* Example: `lakectl dbt delete-branch-schema --branch dbt_jon` will delete the schema associated with the `dbt_jon` branch.
+
+### `merge-branch-schema` Usage
+
+`lakectl dbt merge-branch-schema --source-branch <source branch name> --destination-branch <destination branch name>`: Merge the source schema (associated with the source branch name) to the destination schema (associated with the destination branch name)
+* Example: `lakectl dbt merge-branch-schema --source-branch dbt_jon --destination-branch master` will merge the schema associated with the `dbt_jon` branch to the schema associated with the `master` branch.
+
+## Required changes
+
+In order to implement the new commands, there's a need to map branches to schemas.
```

Comment:
Suppose that we have this mapping, how does the merge work?
Is it something that we already have? Is it the same as `lakectl metastore copy`?

Reply:
Yes it's exactly that